.github/workflows/commit-checker.yml

```diff
@@ -0,0 +1,42 @@
+name: Commit checker
+
+on:
+  pull_request:
+
+jobs:
+  commit-checker:
+    name: Commit checker
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 3
+
+    - name: Get pull-request commits
+      run: |
+        set -x
+        # We are in a merge, HEAD^ is the base branch, HEAD^2 is the merged branch
```

Comment:
"actions/checkout did a merge checkout of the pull-request. As such, the first commit is the merge commit. This means that on HEAD^ is the base branch, and on HEAD^2 are the commits from the pull-request. We now check if those trees have a common parent. If not, we fetch a few more commits till we do. In result, the log between HEAD^ and HEAD^2 will be the commits in the pull-request", or something like that?